### ISSUE TITLE
Accessing a different site collection's Document Library

### ISSUE THREAD

**Issue by Holden15** [pnp/pnpjs#810]
Thank you for reporting an issue, suggesting an enhancement, or asking a question. We appreciate your feedback - to help the team understand your
needs please complete the below template to ensure we have the details to help. Thanks!

**Please check out the [Docs](https://pnp.github.io/pnpjs/) to see if your question is already addressed there. This will help us ensure our documentation covers the most frequent questions.**

### Category
- [ ] Enhancement
- [x] Bug
- [x] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [         ]

Please specify what version(s) of SharePoint you are targeting: [         ]

I want to get a file from a different site collection's document library.  I can pull from the current site collection's document library.

In my service file constructor:

    const kSiteAURL: string = "https://borderstates.sharepoint.com/sites/siteA";

    constructor() {
        let indexOf: number = window.location.pathname.indexOf("/sites/siteA/");
    
        if (indexOf === -1) {
            this._web = new Web(kSiteAURL);
        }
        else {
            this._web = sp.web;
        }
    }

If the app is not in the first site, I create a `Web` object and set it to the first site URL.

In the service call:

    public foobaz(): Promise<Blob> {
        return this._web.getFolderByServerRelativeUrl("/sites/siteA/Test").files.getByName(`Ignite.png`).getBlob();
    }

When I run this from siteB, I get error:

    BranchPlanning | _createDialogError | error:  Error: Error making HttpClient request in queryable [400]  ::> {"odata.error":{"code":"-2147024809, System.ArgumentException","message":{"lang":"en-US","value":"Server relative urls must start with SPWeb.ServerRelativeUrl"}}}

When I run it on siteA, it works fine.  

This works great for when I'm getting data from Lists on siteA from siteB.

What am I doing wrong?

**Comment by patrick-rodgers** [comment]
No entirely sure I follow your code flow, what url are you creating the new Web object with? The Web url needs to match the server relative url of the folder. Meaning you can't have a web object with a url and try to get at a file within another site with that web object even with the correct server relative path. They need to match.

**Comment by Holden15** [comment]
@patrick-rodgers SiteA is where the document library is that I want to access.  During testing, I would add the SPFx app on SiteB.  From SiteB, I wanted to access the library on SiteA because it was easier then reproducing that data on SiteB.

When my services object loads, I check the if the URL is SiteA, if so I set `this._web`.  If the URL is not SiteA, I set `this._web` to SiteA's url to pull the document library.

That logic is fine when working with SP Lists, but throws there error when I try to connect to a document library.
